### PR TITLE
Use Pybind11's find python tool to correctly locate python install information

### LINKED ISSUE #813: Windows: Import error after pip install
## Bug Report

-- edit: removed the checkboxes so that GitHub is no longer confused about open tasks --

-  Incorrect functionality

Describe the bug here. If it is a build issue, please note whether the issue was encountered via `cmake`, `pip`, or `tox`.

## To Reproduce

1. Operating System: Windows 10
2. Python 3.7.9, CMake installed
3. Example snippet that demonstrates the issue:
```python
ImportError: DLL load failed: The specified module could not be found.
```
4. OpenTimelineIO 0.13.0

## Expected Behavior

The import statement to work correctly.

## Logs
```python
Traceback (most recent call last):
  File "C:\Users\User\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "c:\Users\User\.vscode\extensions\ms-python.python-2020.9.114305\pythonFiles\lib\python\debugpy\__main__.py", line 45, in <module>
    cli.main()
  File "c:\Users\User\.vscode\extensions\ms-python.python-2020.9.114305\pythonFiles\lib\python\debugpy/..\debugpy\server\cli.py", line 430, in main
    run()
  File "c:\Users\User\.vscode\extensions\ms-python.python-2020.9.114305\pythonFiles\lib\python\debugpy/..\debugpy\server\cli.py", line 267, in run_file
    runpy.run_path(options.target, run_name=compat.force_str("__main__"))
  File "C:\Users\User\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "E:\Logiciels\Stax\Dev\stax\test_otio.py", line 4, in <module>
    import opentimelineio as otio
  File "e:\Logiciels\Stax\Dev\stax\.venv\lib\site-packages\opentimelineio\__init__.py", line 35, in <module>
    from . import (
  File "e:\Logiciels\Stax\Dev\stax\.venv\lib\site-packages\opentimelineio\opentime.py", line 1, in <module>
    from . _opentime import ( # noqa
ImportError: DLL load failed: The specified module could not be found.
```

### LINKED ISSUE #823: Signed/unsigned comparison
### Incorrect Functionality

```
/opt/OpenTimelineIO/source/src/opentimelineio/imageSequenceReference.cpp: In member function ‘std::string opentimelineio::v1_0::ImageSequenceReference::target_url_for_image_number(int, opentimelineio::v1_0::ErrorStatus*) const’:
/opt/OpenTimelineIO/source/src/opentimelineio/imageSequenceReference.cpp:90:39: warning: comparison of integer expressions of different signedness: ‘std::__cxx11::basic_string<char>::size_type’ {aka ‘long unsigned int’} and ‘const int’ [-Wsign-compare]
   90 |         if (image_num_string.length() <  _frame_zero_padding) {
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
```

### PR DESCRIPTION
**Pull request by Tilix4**
Fixes #813

**Link the Issue(s) this Pull Request is related to.**

Each PR should link at least one issue, in the form:

```Fixes #823```

Use one line for each Issue. This allows auto-closing the related issue when the fix is merged.

Added the pybind11 smart tool to find python: https://pybind11.readthedocs.io/en/latest/compiling.html#findpython-mode

On windows: `/path/to/blender/python -m pip install` wasn't working.

### CODE REVIEW

**Comment by meshula** [comment]
Wow! Fantastic find (so to speak ;) )

File: setup.py [hunk 1]
```diff
@@ -96,7 +96,8 @@ def cmake_generate():
         '-DPYTHON_EXECUTABLE=' + sys.executable,
         '-DOTIO_PYTHON_INSTALL:BOOL=ON',
         '-DOTIO_CXX_INSTALL:BOOL=ON',
-        '-DCMAKE_BUILD_TYPE=' + ('Debug' if _ctx.debug else 'Release')
+        '-DCMAKE_BUILD_TYPE=' + ('Debug' if _ctx.debug else 'Release'),
+        '-DPYBIND11_FINDPYTHON=ON',  # Smart tool to find python's libs
     ]
```
Comment (meshula) [inline setup.py:101]: I'm glad you worked this out ~ I hadn't come across this yet. Would you mind adding a single comment here pointing us back to the documentation, just a link to  https://pybind11.readthedocs.io/en/latest/compiling.html#findpython-mode would be fine

**Review by meshula — CHANGES_REQUESTED** [review]
It's a relief to see that the fix is so simple, after all the research and experimentation it took to get here!

File: setup.py (hunk 1 above)
Comment (Tilix4) [reply setup.py:101]: Done!

**Comment by Tilix4** [comment]
Yes, I'm happy it ends up like this :)

**Comment by Tilix4** [comment]
Don't forget to merge with squashing to avoid cosmetic commits.

**Review by meshula — APPROVED** [review]

**Comment by meshula** [comment]
Just need to clear the checks. Looks like the comment I requested triggered a line length error....!

```
check-manifest succeeded
13
./setup.py:100:89: E501 line too long (146 > 88 characters)
14
make: *** [Makefile:141: lint] Error 1
15

```

**Comment by ssteinbach** [comment]
I edited the title of the PR to reflect what it is doing (to make generating the change notes for the future release easier)

**Comment by meshula** [comment]
Thanks!

**Comment by meshula** [comment]
@Tilix4 We've just found a regression. If one creates a python2.7 conda environment, but python 3.8 is the default python in your path, pybind11 now finds the default python 3.8, not the python that is first in the path. In other words, if I build, using python 2.7, pybind11 prefers python 3.8. It looks like a bit more research is required....! Prior to this commit, pybind11 correctly picks the active python 2.7, or 3.8 as appropriate.

**Comment by meshula** [comment]
Reading the documentation around FindPython in the pybind11 documents is not encouraging - it says things like "with virtualenv/venv support, and Conda support, this tends to find the correct Python version more often than the old system did".

**Comment by Tilix4** [comment]
@meshula Okay, I'll have a look. First of all I'll try to reproduce the issue.

**Comment by Tilix4** [comment]
@meshula I'm sorry but I cannot reproduce this issue.

I've installed a 2.7 conda env: https://repo.anaconda.com/archive/Anaconda2-2019.10-Windows-x86_64.exe

I've tested what follows with both python 3.7 and recently (today) installed 3.8:
- In my usual `cmd` terminal, `python` targets to the 3.8.
- Running the conda env (base): `python` correctly targets the 2.7 python and `pip install` of the current state of master branch correctly installs OTIO to the python 2.7.

If you may provide a `how to reproduce` it'd be very helpful!

> "with virtualenv/venv support, and Conda support, this tends to find the correct Python version more often than the old system did"

Maybe I don't understand this quote correctly but if FindPython is what we are using, it seems to be the appropriate solution.

**Comment by meshula** [comment]
I did almost the same as you, except the inverse. My conda base is 3.8, and I created a 2.7 env... I deleted my otio27 environment and reconstructed it from scratch, and now the problem does not recur. I'll double check with the user who reported the issue.

Thanks for following up, much appreciated.

**Comment by rogernelson** [comment]
Hey all, I have am actually having this issue.  I have python 3.9 install in my /usr/local directory (on Mac). When I invoke setup.py from the system Python (2.7).  I end up with python bindings built for 3.9 in my python 2.7 site packages and the build is not useable.  I can confirm that when I remove the `PYBIND11_FINDPYTHON` option it starts working again.  

You can see my full investigation in these Slack threads:
https://academysoftwarefdn.slack.com/archives/CMQ9J4BQC/p1623277067150200
https://academysoftwarefdn.slack.com/archives/CMQ9J4BQC/p1623280454161300

If you want me to run any tests on my machine let me know.  I will try seeing if replacing both `PYTHON_EXECUTABLE` and `PYBIND11_FINDPYTHON` with `PYBIND11_PYTHON_VERSION` set to the version of the interpreter than invokes the setup.py script might fix the problem.  I am hoping this will work like the findpython option except will match on a particular option.

**Comment by rogernelson** [comment]
I can confirm that using `PYBIND11_PYTHON_VERSION` fixes my problem.  Would someone in this thread be willing to test if it fixes the original issue too?  

You can find my changes here:
https://github.com/rogernelson/OpenTimelineIO/tree/autodesk/nelsonr/python_version

**Pull request merged**